### PR TITLE
Add text metrics to asr eval

### PR DESCRIPTION
**Pull request by stevehuang52**
# What does this PR do ?

Add some text metrics to ASR evaluator to support AST models

### CODE REVIEW

File: nemo/collections/asr/parts/utils/eval_utils.py [hunk 1]
```diff
@@ -15,10 +15,18 @@
 import json
 from typing import Optional, Tuple, Union
 
+from torchmetrics.text import SacreBLEUScore
```
Comment (titu1994) [inline nemo/collections/asr/parts/utils/eval_utils.py:18]: These need to be guarded because we don't always install sacrebleu and rouge

Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/eval_utils.py:18]: won't the dependencies be installed when installing torchmetrics? I didn't know where torchmetrics requires sacrebleu and rouge, thought it uses its own implementaion

Comment (titu1994) [reply nemo/collections/asr/parts/utils/eval_utils.py:18]: I don't want to take a chance with those two. They are annoying libraries to install on other machines than Linux

**Comment by stevehuang52** [comment]
jenkins

File: nemo/collections/asr/parts/utils/eval_utils.py (hunk 1 above)
Comment (stevehuang52) [reply nemo/collections/asr/parts/utils/eval_utils.py:18]: I searched through torchmetrics dependencies and didn't find sacrebleu or rouge. I also looked into its implementation, the SacreBLEUScore implementation doesn't need other dependencies, while the ROUGEScore implementation only needs NLTK, so I think it's safe to add those two metrics?

**Review by titu1994 — COMMENTED** [review]
Nltk has issues on clusters, I'll send you a thread on slack later. Make the class with imports nltk run only inside of the place that's needed, not at the top of the file.

File: nemo/collections/asr/parts/utils/eval_utils.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/asr/parts/utils/eval_utils.py:18]: Nltk has issues on clusters, I'll send you a thread on slack later. Make the class with imports nltk run only inside of the place that's needed, not at the top of the file.

**Review by titu1994 — APPROVED** [review]
Nevermind above comment, nltk itself is fine, it was another library that was acting poorly by installing nltk stuff at import time. 
Lgtm

File: nemo/collections/asr/parts/utils/eval_utils.py (hunk 1 above)
Comment (titu1994) [reply nemo/collections/asr/parts/utils/eval_utils.py:18]: Nvm above. Nltk itself is fine.

**Comment by titu1994** [comment]
jenkins

**Pull request merged**